aten/src/ATen/native/Distance.cpp

```diff
@@ -75,4 +88,269 @@ Tensor cosine_similarity(const Tensor& x1, const Tensor& x2, int64_t dim, double
   return w12.div_((w1 * w2).clamp_min_(eps));
 }
 
+template <typename scalar_t>
+static void to_squareform_frame(
```

Comment:
can you put all the to_squareform and from_squareform things together, respectively?  This is unnecessarily difficult to read.